docs/advanced/language-extensions.md

```diff
@@ -60,3 +60,79 @@ function myFunc(self)
 end
 foo, four = myFunc(nil)
 ```
+
+## Operator Map Types
+
+Lua supports overloading operators on types using [metatable methods](https://www.lua.org/manual/5.4/manual.html#2.4) such as `__add`. But, Javascript and Typescript do not support this. In order to use overloaded operators on types that support them, you can declare special mapping functions in TS that will translate to those operators in Lua.
+
+A common example of an overloaded operator is addition of a mathematical vector type:
+
+```ts
+// Vector type supplied by a library, which supports math operators
+declare interface Vector {
+  x: number;
+  y: number;
+}
+
+declare const a: Vector;
+declare const b: Vector;
+const result = a + b; // Not allowed in TS
+```
+
+To support addition for this type, you can declare a special function:
+
+```ts
+declare const addVector: LuaAddition<Vector, Vector, Vector>;
+const result = addVector(a, b); // transpiles to 'result = a + b'
+```
+
+The mapping function does not have to be declared as global. For example, you could use declaration merging to declare it as a static function on `Vector`:
+
+```ts
+declare namespace Vector {
+  export const add: LuaAddition<Vector, Vector, Vector>;
+}
+
+const result = Vector.add(a, b); // result = a + b
+```
+
+There are also special variants for the mapping types that allow you do declare operator overloads as methods:
+
+```ts
+declare interface Vector {
+  add: LuaAdditionMethod<Vector, Vector>;
+}
+const result = a.add(b); // result = a + b
+```
+
+Some operators may have a different return type based on their inputs. You can support this by using intersection types. For example, our `Vector` type might overload the multiplication operator to scale by a number, or perform a dot product on two `Vectors`:
+
+```ts
+declare namespace Vector {
+  export const mul: LuaMultiplication<Vector, Vector, number> & LuaMultiplication<Vector, number, Vector>;
+}
+
+const dot = Vector.mul(a, b);
```

Comment:
Might be nice to add types to these consts to demonstrate the point, ie `const dot: number = ` and `const scaled: Vector =`